base/genimport/importer.go

```diff
@@ -269,14 +275,108 @@ func removeAllFilesInDirExcept(o *Output, dir string, except_list []string) {
 }
 
 func createPluginGoModFile(o *Output, pkgpath string, dir string) string {
+	newLines := [][]byte{
+		[]byte("module gomacro.imports/" + pkgpath + "\n"),
+	}
+
+	// Attempt to use the local module if present.
+ 	// This only works if the import shares a mod file with current working
+	// directory, because we only know to guess "." for the local location.
+	// TODO: Find a way to support imports from local disk that aren't in
+	//  the cwd project.
+	if pkgModFileInfo, err := getModuleFileInfo("."); err == nil &&
+		(pkgpath == pkgModFileInfo.Path || strings.HasPrefix(pkgpath, pkgModFileInfo.Path + "/")) {
+
+		o.Debugf("importing %s from local %s", pkgpath, pkgModFileInfo.GoMod)
+		newLines = append(newLines, goModReplacementDirectives(o, pkgModFileInfo)...)
+	}
+
+	newGoModText := bytes.Join(newLines, nil)
+
 	gomod := paths.Subdir(dir, "go.mod")
-	err := ioutil.WriteFile(gomod, []byte("module gomacro.imports/"+pkgpath+"\n"), os.FileMode(0644))
+	err := ioutil.WriteFile(gomod, newGoModText, os.FileMode(0o644))
 	if err != nil {
 		o.Errorf("error writing file %q: %v", gomod, err)
 	}
+
 	return gomod
 }
 
+// goModReplacementDirectives will create the replacement directives associated
+// with a module that can be found locally, for the purpose of use in a
+// gomacro.imports mod file.
+func goModReplacementDirectives(o *Output, pkgModFileInfo modInfo) [][]byte {
+	m, err := getModuleFile(pkgModFileInfo)
+	if err != nil {
+		o.Errorf("error getting go.mod", err)
+		return [][]byte{}
+	}
+
+	// Replace the current module with the right directory
+	replaceDirectives := [][]byte{
+		[]byte("replace "+ pkgModFileInfo.Path+" => " + pkgModFileInfo.Dir + "\n"),
+	}
+
+	// Copy the replace directives from the imported mod, so the functionality
+	// remains similar.
+	for _, replaceDirective := range m.Replace {
+		// If the replace directive is to the local disk but not absolute,
+		// point to the correctly location.
+		if modfile.IsDirectoryPath(replaceDirective.New.Path) &&
+			!filepath.IsAbs(replaceDirective.New.Path) {
+
+			replaceDirectives = append(replaceDirectives,
+				[]byte("replace " + replaceDirective.Old.Path + " => " +
+					filepath.Join(pkgModFileInfo.Dir, replaceDirective.New.Path) + "\n"))
+		} else {
+			replaceDirectives = append(replaceDirectives,
+				[]byte("replace " + replaceDirective.Old.Path + " => " +
+					replaceDirective.New.Path + " " + replaceDirective.New.Version + "\n"))
+		}
```

Comment:
I think combining the shared parts of these might make things clearer!
```suggestion
		target := replaceDirective.New.Path + " " + replaceDirective.New.Version
		if modfile.IsDirectoryPath(replaceDirective.New.Path) &&
			!filepath.IsAbs(replaceDirective.New.Path) {
			target = filepath.Join(...)
		}
		replaceDirectives = append(replaceDirectives,
			[]byte("replace " + replaceDirective.Old.Path + " => " + target + "\n"))
```